src/client-agent/buffer.c

```diff
@@ -266,3 +274,111 @@ int w_agentd_get_buffer_lenght() {
 
     return retval;
 }
+
+void w_agentd_buffer_free(unsigned int current_capacity) {
+    w_mutex_lock(&mutex_lock);
+
+    // Ensure the buffer is actually allocated before trying to free.
+    if ( buffer == NULL || current_capacity == 0) {
+        mwarn("Buffer is already unallocated or invalid. Skipping free operation.");
+        w_mutex_unlock(&mutex_lock);
+        return;
+    }
+
+    mdebug2("Freeing the client-buffer.");
+    for ( int i=0; i <= current_capacity; i++) {
+        if (buffer[i] != NULL ) os_free(buffer[i]);
+    }
+    os_free(buffer);
+
+    agt->buflength = 0;
+    i = 0;
+    j = 0;
+
+    // Signal to end the dispatch_buffer thread.
+    w_cond_signal(&cond_no_empty);
+    w_mutex_unlock(&mutex_lock);
+
+    minfo("Client buffer freed successfully.");
+}
+
+int w_agentd_buffer_resize(unsigned int current_capacity, unsigned int desired_capacity) {
+
+    if (desired_capacity <= 0) {
+        merror("Invalid new buffer capacity requested: %u.", desired_capacity);
+        return -1;
+    }
+
+    if (desired_capacity == current_capacity) {
+        return 0;
+    }
+
+    // Attempt to reallocate the buffer
+    unsigned int agent_msg_count = w_agentd_get_buffer_lenght();
+    w_mutex_lock(&mutex_lock);
+
+    char **temp_buffer = NULL;
+    if (desired_capacity > current_capacity) {
+        // We add +1 to the desired capacity for internal management of the circular buffer,
+        // allowing it to distinguish between full and empty states.
+        os_calloc(desired_capacity+1, sizeof(char *), temp_buffer);
+
+        // Copy data in logical order to the new buffer
+        if (j < i ) {
```

Comment:
```suggestion
        os_calloc(desired_capacity + 1, sizeof(char *), temp_buffer);

        // Copy data in logical order to the new buffer
        if (j < i) {
```

Check that the rest of the code follows the style guide correctly: [Coding style guide (C)](https://github.com/wazuh/wazuh/wiki/Coding-style-guide-%28C%29)